### PR TITLE
Persistence: Expose Relational Lock provider as a service

### PR DESCRIPTION
**Pull request by prasar-ashutosh**
#### What type of PR is this?

- Improvement

#### What does this PR do / why is it needed ?
Persistence: Expose Relational Lock provider as a service

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
NO

### CODE REVIEW

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    public void createAndInitialize(Executor<SqlGen, TabularData, SqlPlan> executor, LockInfoDataset lockDataset, String tableName)
+    {
+        executor.setSqlLogging(sqlLogging());
+        LOGGER.info("Creating and initializing Lock dataset: " + lockDataset.name());
+        Map<String, PlaceholderValue> placeHolderKeyValues = new HashMap<>();
+        placeHolderKeyValues.put(BATCH_START_TS_PATTERN, PlaceholderValue.of(LocalDateTime.now(executionTimestampClock()).format(DATE_TIME_FORMATTER), false));
+        executor.executePhysicalPlan(createAndInitializeLockSqlPlan(lockDataset, tableName), placeHolderKeyValues);
+    }
+
+    public void lock(Executor<SqlGen, TabularData, SqlPlan> executor, LockInfoDataset lockDataset)
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java:99]: Shall we just call this acquireLock?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    // ---------- Private Fields ----------
+    private static final Logger LOGGER = LoggerFactory.getLogger(RelationalLockProvider.class);
+
+    // ------API-----
+    public void createAndInitialize(Executor<SqlGen, TabularData, SqlPlan> executor, LockInfoDataset lockDataset, String tableName)
+    {
+        executor.setSqlLogging(sqlLogging());
+        LOGGER.info("Creating and initializing Lock dataset: " + lockDataset.name());
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java:93]: Logging happening before case conversion, so we might see a mismatch between the logged name and the actual name?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        Map<String, PlaceholderValue> placeHolderKeyValues = new HashMap<>();
+        placeHolderKeyValues.put(BATCH_START_TS_PATTERN, PlaceholderValue.of(LocalDateTime.now(executionTimestampClock()).format(DATE_TIME_FORMATTER), false));
+        executor.executePhysicalPlan(createAndInitializeLockSqlPlan(lockDataset, tableName), placeHolderKeyValues);
+    }
+
+    public void lock(Executor<SqlGen, TabularData, SqlPlan> executor, LockInfoDataset lockDataset)
+    {
+        executor.setSqlLogging(sqlLogging());
+        LOGGER.info("Acquiring lock using LockInfoDataset: " + lockDataset.name());
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java:102]: Same as above - Logging happening before case conversion, so we might see a mismatch between the logged name and the actual name?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/test/java/org/finos/legend/engine/persistence/components/ingestmode/mixed/TestRunner.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.finos.legend.engine.persistence.components.executor.Executor;
+import org.finos.legend.engine.persistence.components.relational.api.RelationalIngestor;
+import org.finos.legend.engine.persistence.components.relational.api.RelationalLockProvider;
+import org.finos.legend.engine.persistence.components.relational.h2.H2Sink;
+import org.finos.legend.engine.persistence.components.relational.jdbc.JdbcConnection;
+import org.finos.legend.engine.persistence.components.relational.jdbc.JdbcHelper;
+import org.finos.legend.engine.persistence.components.util.LockInfoDataset;
+
+public class TestRunner implements Runnable
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/test/java/org/finos/legend/engine/persistence/components/ingestmode/mixed/TestRunner.java:25]: Shall we make the name of the class a bit more specific?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java (hunk 3 above)
Comment (prasar-ashutosh) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java:102]: I think it's fine, will be expensive to convert this just for logging. It represents the value that user passed.
Will print the caseConversion param here.

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java (hunk 2 above)
Comment (prasar-ashutosh) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalLockProviderAbstract.java:93]: I think it's fine, will be expensive to convert this just for logging. It represents the value that user passed.
Will print the caseConversion param here.

**Review by epsstan — APPROVED** [review]

**Pull request merged**